### ISSUE TITLE
ty doesnt respect UV_PROJECT_ENVIRONMENT or `uv run`, leading it to incorrectly claim `unresolved-import`

### ISSUE THREAD

**Issue by brycedrennan** [astral-sh/ty#1733]
In a docker image, when `UV_PROJECT_ENVIRONMENT=/usr/local`, `ty` cannot resolve third-party imports even though `uv run python` can import them just fine.

The use case is having uv install into vanilla python docker images and then running the type checker against code in that image.

I see in previous discussion that we dont want to complicate `ty` with all the logic of finding the right python installation, but I would have expected `uv run` to have solved that such that `uv run ty` "just works"

Also note that `ty` is not respecting the environment that it's installed into. This is contrary to a [recent comment:](https://github.com/astral-sh/ty/issues/684#issuecomment-3533238411) 
> We now respect the Python environment that ty itself is installed in

**Repro**
```bash
docker build -t ty-bug-repro .

# ty claims unresolved import:
docker run --rm ty-bug-repro uv run ty check
# error[unresolved-import]: Cannot resolve imported module `requests`

docker run --rm ty-bug-repro ty check
# error[unresolved-import]: Cannot resolve imported module `requests`

# But Python import works:
docker run --rm ty-bug-repro uv run main.py
# /usr/local/lib/python3.11/site-packages/requests/__init__.py
```

**Dockerfile:**
```dockerfile
FROM python:3.11-slim
COPY --from=ghcr.io/astral-sh/uv:latest /uv /bin/
ENV UV_PROJECT_ENVIRONMENT=/usr/local
WORKDIR /app
COPY . .
RUN uv sync 
```

**pyproject.toml:**
```toml
[project]
name = "ty-bug-repro"
version = "0.1.0"
requires-python = ">=3.11"
dependencies = ["requests", "ty"]
```

**main.py:**
```python
import requests
print(requests.__file__)
```
 

**Workarounds**
I don't love any of these. Mostly because every other command in my makefile is identical for local dev vs running in my docker image.

**1. Set PYTHONPATH:**
```dockerfile
ENV PYTHONPATH=/usr/local/lib/python3.11/site-packages
```

**2. Pass --python to ty:**
```bash
ty check --python /usr/local/bin/python3
```

**3. Fake a venv:**
```dockerfile
ENV VIRTUAL_ENV=/usr/local
RUN printf "home = /usr/local/bin\nversion = 3.11\n" > /usr/local/pyvenv.cfg
```

**4. Suppress the error**
```toml
[tool.ty.rules]
unresolved-import = "ignore"
```

### Version

0.0.1-alpha.29

**Comment by zanieb** [comment]
I'm surprised to hear that `uv run ty check` doesn't work with `UV_PROJECT_ENVIRONMENT=/usr/local`. I can look into that.

**Comment by notarealuseralcemy** [comment]
```
ty check --python $(uv run -- which python)
```

is maybe a slightly easier workaround in *nix, if the command should be run in multiple environments (like CI/pre-commit hooks, in my case).

**Comment by zanieb** [comment]
The reason it doesn't work with `UV_PROJECT_ENVIRONMENT` is that there's no `pyvenv.cfg` in the `VIRTUAL_ENV` location and ty rejects it. The "fix" is probably to allow environments via `VIRTUAL_ENV` as long as they have the expected site-packages structure.

**Comment by carljm** [comment]
@zanieb We should also make the same change for "environment ty is installed into" -- currently that requires venv only, but it should allow system environments.

**Comment by zanieb** [comment]
We might not actually want `VIRTUAL_ENV=<system-env>` to work, as it could be confusing? Maybe we should read `UV_PROJECT_ENVIRONMENT` and only allow it in that case?

**Comment by ericfitz** [comment]
I don't want to create a new issue because I think that this change would resolve my issue.

I use ty as my python linter in VS Code.

uv did such a good job of managing venvs that I only run python scripts with uv run, and I put a `/// script dependencies...` block in all my python scripts.  But the side effect is that in all my projects, I get unresolved import and reference warnings.

Does this need to be a separate issue, or will the fix for this issue resolve my problem as well, or do you also need a VS code setting when ty is set to be a python linter?

**Comment by carljm** [comment]
@ericfitz this issue is specifically about the case where `UV_PROJECT_ENVIRONMENT` is pointing to a "system" Python (typically because it's used in a Docker container). It's not clear if that's your issue? It sounds like maybe your issue is #691 ?

**Comment by twsl** [comment]
+1
i want to add that setting the path explicitly in the pyproject.toml doesnt work for me, as in ci pipelines i use a venv while in local dev or devcontainer i use system env
